app/src/main/java/org/mozilla/tv/firefox/utils/ServiceLocator.kt

```diff
@@ -85,4 +85,17 @@ open class ServiceLocator(val app: Application) {
     }
     open val sessionRepo by lazy { SessionRepo(sessionManager, sessionUseCases, turboMode).apply { observeSources() } }
     open val settingsRepo by lazy { SettingsRepo(app) }
+
+    // The first intent the App was launched with.  Used to pass configuration through to Gecko.
+    lateinit var launchSafeIntent: SafeIntent
+
+    fun notifyLaunchWithSafeIntent(safeIntent: SafeIntent): Boolean {
```

Comment:
I'd prefer this to be in `WebRenderComponents` than in `ServiceLocator`. Alternatively you can create companion class in `SafeIntent`

Reply:
I will try to make this so.